zerver/models.py

```diff
@@ -1827,3 +1833,22 @@ def get_bot_services(user_profile_id):
 def get_service_profile(email, realm, service_name):
     # type: (str, Realm, str) -> Service
     return Service.objects.get(user_profile__email=email, user_profile__realm=realm, name=service_name)
+
+def get_slash_commands_by_realm(realm, is_active=None):
+    # type: (Realm, Optional[bool]) -> List[Any]
+    if is_active is None:
+        users = list(UserProfile.objects.filter(realm=realm, bot_type=UserProfile.SLASH_COMMANDS).values('email'))
+    else:
+        users = list(UserProfile.objects.filter(realm=realm, bot_type=UserProfile.SLASH_COMMANDS,
+                                                is_active=is_active).values('email', 'is_active'))
+    usernames = [email_to_username(user['email']) for user in users]
+    slash_commands = []
+    for username in usernames:
+        if username[-4:] != "-bot":
+            raise Exception("Email id is not in standard bot form.")
```

Comment:
Are you requiring slash command emails to have the form `<command>-bot@domain`?  It looks like this might cause conflicts with other types of bots.